src/kili/graphql/graphql_client.py

```diff
@@ -60,7 +63,59 @@ def __init__(
             method="POST",
             # can add other requests kwargs here
         )
-        self._gql_client = Client(transport=gql_transport, fetch_schema_from_transport=True)
+
+        graphql_schema_path = self._get_graphql_schema_path()
+        self._cache_graphql_schema(graphql_schema_path)
+
+        self._gql_client = Client(
+            schema=graphql_schema_path.read_text(encoding="utf-8"),
+            transport=self.gql_transport,
+        )
+
+    def _cache_graphql_schema(self, graphql_schema_path: Path) -> None:
+        """
+        Cache the graphql schema (if not already in cache).
+
+        If the schema is not in cache, it will be fetched from the server.
+
+        Also deletes old cache files.
+        """
+        for old_cache_file in graphql_schema_path.parent.glob("*.graphql"):
+            if old_cache_file.name != graphql_schema_path.name:
+                old_cache_file.unlink()
+
+        if graphql_schema_path.is_file() and graphql_schema_path.stat().st_size > 0:
+            return
+
+        with Client(transport=self.gql_transport, fetch_schema_from_transport=True) as session:
+            schema_str = print_schema(session.client.schema)  # type: ignore
+
+        graphql_schema_path.parent.mkdir(parents=True, exist_ok=True)
+
+        with graphql_schema_path.open("w", encoding="utf-8") as file:
+            file.write(schema_str)
+
+    def _get_graphql_schema_path(self) -> Path:
+        """
+        Get the path of the GraphQL schema
+        """
+        endpoint_netloc = urlparse(self.endpoint).netloc
+        version = self.get_kili_app_version()
+        timestamp = datetime.now().strftime(r"%Y%m%d")  # cache for one day at most
```

Comment:
it won't be a 1 day cache if it is generated at 23:59:59. It would be better to index the cache by the timestamp in seconds, and check if the cache is valid depending on this timestamp being more than 1 day-old.